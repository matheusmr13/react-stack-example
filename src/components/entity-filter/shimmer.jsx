import React from 'react';
import FieldShimmer from 'components/field-filter/shimmer';

const EntityFilterShimmer = () => Array(4).fill(0).map((e, i) => <FieldShimmer key={i} />);

export default EntityFilterShimmer;
