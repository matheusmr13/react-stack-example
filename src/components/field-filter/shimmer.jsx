import React from 'react';
import ContentLoader from 'react-content-loader';

const FieldFilterShimmer = () => (
	<ContentLoader
		height={80}
		width={250}
		speed={2}
		primaryColor="#f3f3f3"
		secondaryColor="#ecebeb"
	>
		<rect x="10" y="38" rx="5" ry="5" width="230.3" height="26.6" />
		<rect x="10" y="10.27" rx="4" ry="4" width="121.01" height="16.77" />
	</ContentLoader>
);

export default FieldFilterShimmer;
