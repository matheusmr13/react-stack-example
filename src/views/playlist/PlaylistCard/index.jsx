import React from 'react';

import './PlaylistCard.css';

const PlaylistCard = ({ playlist }) => {
	return (
		<div className="playlist-card">
			<img src={playlist.images[0].url} alt={playlist.name} />
			<div className="playlist-card__name">{playlist.name}</div>
		</div>
	);
};

export default PlaylistCard;
