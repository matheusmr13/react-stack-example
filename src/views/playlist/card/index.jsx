import React from 'react';
import PropTypes from 'prop-types';
import PlaylistPropType from 'services/playlist/proptype';

const PlaylistCard = ({ playlist }) => (
	<div>
		<img src={playlist.images[0].url} alt={playlist.name} />
		<div>{playlist.name}</div>
	</div>
);

PlaylistCard.propTypes = {
	playlist: PropTypes.shape(PlaylistPropType).isRequired
};

export default PlaylistCard;
