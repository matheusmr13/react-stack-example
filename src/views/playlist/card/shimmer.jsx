import React from 'react';
import ContentLoader from 'react-content-loader';

const CardShimmer = () => (
	<ContentLoader
		height={337}
		width={300}
		speed={2}
		primaryColor="#282828"
		secondaryColor="#404040"
	>
		<rect x="1" y="1" rx="5" ry="5" width="298" height="298" />
		<rect x="50" y="310" rx="5" ry="5" width="200" height="30" />
	</ContentLoader>
);

export default CardShimmer;
