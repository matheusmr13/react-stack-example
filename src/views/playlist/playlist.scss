.playlists {
	background: linear-gradient(0deg,#1db954,#000);
	min-height: calc(100vh - 64px);

	@media screen and (max-width: 568px) {
		text-align: center;
	}

	&__header {
		padding: 12px;
		display: flex;
		justify-content: space-between;
		@media screen and (max-width: 568px) {
			flex-direction: column-reverse;
		}
	}

	&__content {
		margin: auto;
		padding: 24px;
		max-width: calc((300px + 24px) * 4);
	}

	&__message {
		font-size: 30px;
		color: #fff;
		font-weight: bold;
		display: flex;
		align-items: center;

		@media screen and (max-width: 568px) {
			margin-top: 32px;
			width: 300px;
			margin-left: auto;
			margin-right: auto;
		}
	}

	&__logged-user {
		font-size: 30px;
		color: #fff;
		font-weight: bold;
	}

	&__filter {
		padding-left: 12px;
		width: 300px;

		@media screen and (max-width: 568px) {
			margin: auto;
		}

		input {
			color: #fff !important;
		}
	}

	&__list-container {
		text-align: center;
	}

	&__empty {
		color: gray;
		font-size: 24px;
		text-align: center;
	}

}