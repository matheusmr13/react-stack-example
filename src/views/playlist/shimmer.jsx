import React from 'react';
import ContentLoader from 'react-content-loader';
import PlaylistCardShimmer from './card/shimmer';

export const PlaylistsShimmer = () => Array(4).fill(0).map((e, i) => (
	<div className="playlist-card" key={i}>
		<PlaylistCardShimmer />
	</div>
));

export const PlaylistNameShimmer = () => (
	<div style={{ minWidth: '260px' }}>
		<ContentLoader
			height={32}
			width={300}
			speed={2}
			primaryColor="#282828"
			secondaryColor="#404040"
		>
			<rect x="1" y="1" rx="5" ry="5" width="298" height="30" />
		</ContentLoader>
	</div>
);
