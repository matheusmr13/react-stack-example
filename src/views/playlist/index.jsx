import React, { Component } from 'react';
import PropTypes from 'prop-types';
import PlaylistPropType from 'services/playlist/proptype';

import UserInfo from 'components/user-info';

import TextField from 'material-ui/TextField';

import PlaylistCard from './card';

import './playlist.css';

class Playlists extends Component {
	state = {
		nameFilter: '',
		filteredPlaylists: []
	}

	componentDidMount() {
		this.props.fetchInitialPlaylists();
	}

	componentWillReceiveProps(props) {
		if (props.playlists !== this.props.playlists) {
			this.filterPlaylists(props.playlists);
		}
	}

	onFilterByName = (e, nameFilter) => {
		this.setState({
			nameFilter
		}, () => {
			this.filterPlaylists(this.props.playlists);
		});
	}

	filterPlaylists(playlists) {
		const { nameFilter } = this.state;
		this.setState({
			filteredPlaylists: playlists.playlists.items
				.filter(playlist => playlist.name
					.toLowerCase()
					.indexOf(nameFilter.toLowerCase()) > -1)
		});
	}

	render() {
		const { loadingPlaylists, playlists, loggedUser } = this.props;
		const { nameFilter, filteredPlaylists } = this.state;

		if (loadingPlaylists) {
			return 'Loading';
		}
		return (
			<div className="playlists">
				<div className="playlists__content">
					<div className="playlists__header">
						<div className="playlists__message">{playlists.message}</div>
						<UserInfo user={loggedUser} onLogout={this.props.onLogout} />
					</div>
					<div className="playlists__filter">
						<TextField
							hintText="Filter"
							onChange={this.onFilterByName}
							value={nameFilter}
							fullWidth
						/>
					</div>
					<div className="playlists__list-container">
						<div className="playlists__list">
							{
								filteredPlaylists
									.map(playlist => (
										<PlaylistCard
											key={playlist.id}
											playlist={playlist}
										/>
									))
							}
						</div>
					</div>
				</div>
			</div>
		);
	}
}

Playlists.defaultProps = {
	playlists: {},
	loggedUser: null
};

Playlists.propTypes = {
	loggedUser: PropTypes.shape({
		info: PropTypes.shape({
			display_name: PropTypes.string
		})
	}),
	fetchInitialPlaylists: PropTypes.func.isRequired,
	onLogout: PropTypes.func.isRequired,
	loadingPlaylists: PropTypes.bool.isRequired,
	playlists: PropTypes.shape({
		message: PropTypes.string.isRequired,
		playlists: PropTypes.shape({
			items: PropTypes.arrayOf(PropTypes.shape(PlaylistPropType)).isRequired
		}).isRequired
	})
};

export default Playlists;
