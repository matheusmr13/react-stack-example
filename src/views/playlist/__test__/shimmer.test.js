import React from 'react';
import { shallow } from 'enzyme';
import { PlaylistsShimmer, PlaylistNameShimmer } from 'views/playlist/shimmer';
import PlaylistCardShimmer from 'views/playlist/card/shimmer';

describe('PlaylistsShimmer component', () => {
	it('should render list shimmer properly', () => {
		const playlist = shallow(<PlaylistsShimmer />);
		expect(playlist.find(PlaylistCardShimmer)).toHaveLength(4);
		expect(playlist).toMatchSnapshot();
	});
	it('should render name shimmer properly', () => {
		const playlistName = shallow(<PlaylistNameShimmer />);
		expect(playlistName).toMatchSnapshot();
	});
});
