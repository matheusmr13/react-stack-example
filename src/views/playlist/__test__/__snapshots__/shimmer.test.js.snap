// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistsShimmer component should render list shimmer properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="playlist-card"
>
          <CardShimmer />
</div>,
        <div
          className="playlist-card"
>
          <CardShimmer />
</div>,
        <div
          className="playlist-card"
>
          <CardShimmer />
</div>,
        <div
          className="playlist-card"
>
          <CardShimmer />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <CardShimmer />,
          "className": "playlist-card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <CardShimmer />,
          "className": "playlist-card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <CardShimmer />,
          "className": "playlist-card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": <CardShimmer />,
          "className": "playlist-card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="playlist-card"
>
            <CardShimmer />
</div>,
          <div
            className="playlist-card"
>
            <CardShimmer />
</div>,
          <div
            className="playlist-card"
>
            <CardShimmer />
</div>,
          <div
            className="playlist-card"
>
            <CardShimmer />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <CardShimmer />,
            "className": "playlist-card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <CardShimmer />,
            "className": "playlist-card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <CardShimmer />,
            "className": "playlist-card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <CardShimmer />,
            "className": "playlist-card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PlaylistsShimmer component should render name shimmer properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ContentLoader
        className=""
        height={32}
        preserveAspectRatio="xMidYMid meet"
        primaryColor="#282828"
        secondaryColor="#404040"
        speed={2}
        type="facebook"
        width={300}
>
        <rect
                height="30"
                rx="5"
                ry="5"
                width="298"
                x="1"
                y="1"
        />
</ContentLoader>,
      "style": Object {
        "minWidth": "260px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <rect
          height="30"
          rx="5"
          ry="5"
          width="298"
          x="1"
          y="1"
/>,
        "className": "",
        "height": 32,
        "preserveAspectRatio": "xMidYMid meet",
        "primaryColor": "#282828",
        "secondaryColor": "#404040",
        "speed": 2,
        "type": "facebook",
        "width": 300,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "30",
          "rx": "5",
          "ry": "5",
          "width": "298",
          "x": "1",
          "y": "1",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ContentLoader
          className=""
          height={32}
          preserveAspectRatio="xMidYMid meet"
          primaryColor="#282828"
          secondaryColor="#404040"
          speed={2}
          type="facebook"
          width={300}
>
          <rect
                    height="30"
                    rx="5"
                    ry="5"
                    width="298"
                    x="1"
                    y="1"
          />
</ContentLoader>,
        "style": Object {
          "minWidth": "260px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <rect
            height="30"
            rx="5"
            ry="5"
            width="298"
            x="1"
            y="1"
/>,
          "className": "",
          "height": 32,
          "preserveAspectRatio": "xMidYMid meet",
          "primaryColor": "#282828",
          "secondaryColor": "#404040",
          "speed": 2,
          "type": "facebook",
          "width": 300,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "30",
            "rx": "5",
            "ry": "5",
            "width": "298",
            "x": "1",
            "y": "1",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
